File: RepoTransBench/target_projects/Java/Go/code4craft_hello-design-pattern/tests/original/creational/singleton/hello_world_singleton_full_test.go
package singleton

import (
	"testing"

	"github.com/stretchr/testify/assert"
	"sync"
)

// Singleton implementation for HelloWorldSingleton
type HelloWorldSingleton struct{}

var (
	instance     *HelloWorldSingleton
	once         sync.Once
)

func Instance() *HelloWorldSingleton {
	once.Do(func() {
		instance = &HelloWorldSingleton{}
	})
	return instance
}

func (h *HelloWorldSingleton) HelloWorld() string {
	return "Hello Singleton!"
}

func TestSingletonInstance_ReturnsSameInstance(t *testing.T) {
	inst1 := Instance()
	inst2 := Instance()
	assert.Same(t, inst1, inst2, "Instance() should always return the same instance")
}

func TestSingletonMessage(t *testing.T) {
	assert.Equal(t, "Hello Singleton!", Instance().HelloWorld())
}

File: RepoTransBench/target_projects/Java/C#/code4craft_hello-design-pattern/tests/original/creational/singleton/HelloWorldSingletonFullTest.cs
using Xunit;

namespace HelloDesignPattern.Tests.creational.singleton
{
    public class HelloWorldSingletonFullTest
    {
        [Fact]
        public void TestSingletonInstance_ReturnsSameInstance()
        {
            var instance1 = HelloWorldSingleton.Instance();
            var instance2 = HelloWorldSingleton.Instance();
            Assert.Same(instance1, instance2);
        }

        [Fact]
        public void TestSingletonMessage()
        {
            Assert.Equal("Hello Singleton!", HelloWorldSingleton.Instance().HelloWorld());
        }
    }
}